From 8981cb34f132d1394b1a4158de90a42ec8c5bc25 Mon Sep 17 00:00:00 2001
From: "Mallikarjun, chippalkatti" <mallikarjunx.chippalkatti@intel.com>
Date: Mon, 24 Apr 2017 14:43:23 +0530
Subject: [PATCH 0733/2313] ASoC: Intel: CNL: Retrieve module id from GUID

This patch retrieves module id from GUID for copier module used
for SoundWire BRA feature.

Change-Id: Ib9453f929e4b0280535a2151851ac04c5098c806
Signed-off-by: Mallikarjun, chippalkatti <mallikarjunx.chippalkatti@intel.com>
Signed-off-by: Dharageswari R <dharageswari.r@intel.com>
Reviewed-by: audio_build <audio_build@intel.com>
Reviewed-by: Singh, Guneshwor O <guneshwor.o.singh@intel.com>
Reviewed-by: Prodduvaka, Leoni <leoni.prodduvaka@intel.com>
Reviewed-by: Kp, Jeeja <jeeja.kp@intel.com>
Reviewed-by: Diwakar, Praveen <praveen.diwakar@intel.com>
Tested-by: Avati, Santosh Kumar <santosh.kumar.avati@intel.com>
---
 sound/soc/intel/skylake/skl-messages.c  | 14 ++++----------
 sound/soc/intel/skylake/skl-sst-dsp.h   |  1 +
 sound/soc/intel/skylake/skl-sst-utils.c | 12 ++++++++++++
 3 files changed, 17 insertions(+), 10 deletions(-)

diff --git a/sound/soc/intel/skylake/skl-messages.c b/sound/soc/intel/skylake/skl-messages.c
index 54d45f00..ef8663a 100644
--- a/sound/soc/intel/skylake/skl-messages.c
+++ b/sound/soc/intel/skylake/skl-messages.c
@@ -378,12 +378,9 @@ static int cnl_sdw_bra_pipe_cfg_pb(struct skl_sst *ctx,
 	host_cpr_params.s_fmt = 32;
 	host_cpr_params.linktype = 0;
 	host_cpr_params.stream = 0;
+	host_cpr_cfg.id.module_id = skl_get_module_id(ctx,
+					(uuid_le *)host_cpr_cfg.guid);
 
-#if IS_ENABLED(CONFIG_SND_SOC_INTEL_CNL_FPGA)
-	host_cpr_cfg.id.module_id = 3;
-#else
-	host_cpr_cfg.id.module_id = 4;
-#endif
 	host_cpr_cfg.id.instance_id = 1;
 	host_cpr_cfg.id.pvt_id = skl_get_pvt_id(ctx,
 		(uuid_le *)host_cpr_cfg.guid, host_cpr_cfg.id.instance_id);
@@ -612,12 +609,9 @@ static int cnl_sdw_bra_pipe_cfg_cp(struct skl_sst *ctx,
 	link_cpr_params.s_fmt = 32;
 	link_cpr_params.linktype = 0;
 	link_cpr_params.stream = 0;
+	host_cpr_cfg.id.module_id = skl_get_module_id(ctx,
+					(uuid_le *)host_cpr_cfg.guid);
 
-#if IS_ENABLED(CONFIG_SND_SOC_INTEL_CNL_FPGA)
-	link_cpr_cfg.id.module_id = 3;
-#else
-	link_cpr_cfg.id.module_id = 4;
-#endif
 	link_cpr_cfg.id.instance_id = 3;
 	link_cpr_cfg.id.pvt_id = skl_get_pvt_id(ctx,
 		(uuid_le *)link_cpr_cfg.guid, link_cpr_cfg.id.instance_id);
diff --git a/sound/soc/intel/skylake/skl-sst-dsp.h b/sound/soc/intel/skylake/skl-sst-dsp.h
index f427c93..62881fb 100644
--- a/sound/soc/intel/skylake/skl-sst-dsp.h
+++ b/sound/soc/intel/skylake/skl-sst-dsp.h
@@ -280,6 +280,7 @@ int bxt_load_library(struct sst_dsp *ctx, struct skl_lib_info *linfo, int lib_co
 
 int snd_skl_parse_uuids(struct sst_dsp *ctx, const struct firmware *fw,
 				unsigned int offset, int index);
+int skl_get_module_id(struct skl_sst *ctx, uuid_le *uuid_mod);
 int skl_get_pvt_id(struct skl_sst *ctx, uuid_le *uuid_mod, int instance_id);
 int skl_put_pvt_id(struct skl_sst *ctx, uuid_le *uuid_mod, int *pvt_id);
 int skl_get_pvt_instance_id_map(struct skl_sst *ctx,
diff --git a/sound/soc/intel/skylake/skl-sst-utils.c b/sound/soc/intel/skylake/skl-sst-utils.c
index ff87eea..19fdd49 100644
--- a/sound/soc/intel/skylake/skl-sst-utils.c
+++ b/sound/soc/intel/skylake/skl-sst-utils.c
@@ -137,6 +137,18 @@ static int skl_get_pvtid_map(struct uuid_module *module, int instance_id)
 	return -EINVAL;
 }
 
+int skl_get_module_id(struct skl_sst *ctx, uuid_le *uuid_mod)
+{
+	struct uuid_module *module;
+
+	list_for_each_entry(module, &ctx->uuid_list, list) {
+		if (uuid_le_cmp(*uuid_mod, module->uuid) == 0)
+			return module->id;
+	}
+	return -EINVAL;
+}
+EXPORT_SYMBOL_GPL(skl_get_module_id);
+
 int skl_get_pvt_instance_id_map(struct skl_sst *ctx,
 				int module_id, int instance_id)
 {
-- 
2.7.4

